samples/test/delete_cluster.test.js

```diff
@@ -0,0 +1,112 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const {assert} = require('chai');
+const cleanup = require('./clean.js');
+const {describe, it, before, after} = require('mocha');
+const {
+  RealmsServiceClient,
+  GameServerClustersServiceClient,
+} = require('@google-cloud/game-servers');
+
+const cp = require('child_process');
+const uuid = require('uuid');
+
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+
+const LOCATION = 'us-central1';
+const GKE_CLUSTER_NAME =
+  process.env.SAMPLE_CLUSTER_NAME ||
+  'projects/1046198160504/locations/us-west1-a/clusters/grpc-bug-cluster';
+
+describe('Game Servers Delete Cluster Test', () => {
+  const realmsClient = new RealmsServiceClient();
+  const gameClustersClient = new GameServerClustersServiceClient();
+  let projectId, realmId, gameClusterId;
+
+  before(async () => {
+    await cleanup();
+
+    // Create a realm
+    projectId = await realmsClient.getProjectId();
+    realmId = `delete-realm-${uuid.v4()}`;
+    gameClusterId = `test-${uuid.v4()}`;
+
+    const createRealmRequest = {
+      parent: `projects/${projectId}/locations/${LOCATION}`,
+      realmId: realmId,
+      realm: {
+        timeZone: 'US/Pacific',
+        description: 'Test Game Server realm',
+      },
+    };
+
+    const [operation1] = await realmsClient.createRealm(createRealmRequest);
+    await operation1.promise();
+
+    // Create a cluster
+    const createClusterRequest = {
+      parent: `projects/${projectId}/locations/${LOCATION}/realms/${realmId}`,
+      gameServerClusterId: gameClusterId,
+      gameServerCluster: {
+        description: 'My Game Server Cluster',
+        connectionInfo: {
+          gkeClusterReference: {
+            // Provide full resource name of a Kubernetes Engine cluster
+            cluster: GKE_CLUSTER_NAME,
+          },
+          namespace: 'default',
+        },
+      },
+    };
+
+    const [operation2] = await gameClustersClient.createGameServerCluster(
+      createClusterRequest
+    );
+    await operation2.promise();
+  });
+
+  it('should delete a Game Server cluster in a realm', async () => {
+    const create_output = execSync(
+      `node delete_cluster.js ${projectId} ${LOCATION} ${realmId} ${gameClusterId}`
+    );
+    assert.match(create_output, /Game Server cluster deleted/);
+  });
+
+  after(async () => {
+    // Delete the Game Server cluster
+    const deleteClusterRequest = {
+      // Provide full resource name of a Game Server Realm
+      name: `projects/${projectId}/locations/${LOCATION}/realms/${realmId}/gameServerClusters/${gameClusterId}`,
+    };
+    try {
+      const [operation1] = await gameClustersClient.deleteGameServerCluster(
+        deleteClusterRequest
+      );
+      await operation1.promise();
+    } catch (ex) {
+      console.log(ex);
```

Comment:
I'd love to eventually get to the bottom of why this is necessary, but seems reasonable for now (we should just keep a close eye and make sure we're not leaking servers).